sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -87,25 +114,30 @@ def __init__(self, state_handler, data_channel_factory):
   def do_instruction(self, request):
     request_type = request.WhichOneof('request')
     if request_type:
-      # E.g. if register is set, this will construct
-      # InstructionResponse(register=self.register(request.register))
-      return beam_fn_api_pb2.InstructionResponse(**{
-          'instruction_id': request.instruction_id,
-          request_type: getattr(self, request_type)
-                        (getattr(request, request_type), request.instruction_id)
-      })
+      # E.g. if register is set, this will call self.register(request.register))
+      return getattr(self, request_type)(
+          getattr(request, request_type), request.instruction_id)
     else:
       raise NotImplementedError
 
-  def register(self, request, unused_instruction_id=None):
+  def register(self, request, instruction_id):
     for process_bundle_descriptor in request.process_bundle_descriptor:
       self.fns[process_bundle_descriptor.id] = process_bundle_descriptor
-    return beam_fn_api_pb2.RegisterResponse()
+    return beam_fn_api_pb2.InstructionResponse(**{
+        'instruction_id': instruction_id,
+        'register': beam_fn_api_pb2.RegisterResponse()})
 
   def process_bundle(self, request, instruction_id):
     bundle_processor.BundleProcessor(
         self.fns[request.process_bundle_descriptor_reference],
         self.state_handler,
         self.data_channel_factory).process_bundle(instruction_id)
 
-    return beam_fn_api_pb2.ProcessBundleResponse()
+    return beam_fn_api_pb2.InstructionResponse(**{
+        'instruction_id': instruction_id,
+        'process_bundle': beam_fn_api_pb2.ProcessBundleResponse()})
+
+  def process_bundle_progress(self, request, instruction_id):
+    return beam_fn_api_pb2.InstructionResponse(**{
+        'instruction_id': instruction_id,
+        'error': 'Not Supported'})
```

Comment:
Any reason you didn't raise an error here rather than change all the signatures?

Reply:
The `error` field in `InstructionResponse` is shared by all types of responses. This signature change allows the instruction handlers to send error messages that the runner understands, unlike before where we always send a traceback.

Reply:
Hmm... I didn't think the runner was suppose to interpret this error message, just look for its presence. 

I'm fine with this if you'd prefer, though in that case change all instance of (**{'key': value}) to (key=value) which can be used now.

Reply:
Thanks for the cleanup!!